src/atopile/front_end.py

```diff
@@ -1197,18 +1294,29 @@ def visitDeclaration_stmt(self, ctx: ap.Declaration_stmtContext) -> KeyOptMap:
             )
 
         assigned_name = assigned_value_ref[0]
-        unit = self._get_unit_from_ctx(ctx.type_info().name_or_attr())
+        unit_ctx: ap.Name_or_attrContext = ctx.type_info().name_or_attr()
+        # TODO: @v0.4.0: remove this shim
+        unit_ref = self.visitName_or_attr(unit_ctx)
```

Comment:
I dislike the declaration syntax atm, which is why I've marked this as deprecated

Should consider how we can expand to domains etc... too